### PR TITLE
CLOUD-4781: Use Jenkins Configuration As Code Plugin

### PR DESCRIPTION
**Pull request by r-kok**
JCasC is now the more or less official way to program your Jenkins:

https://www.jenkins.io/projects/jcasc/

### CODE REVIEW

File: casc_configs/jenkins.yaml [hunk 1]
```diff
@@ -0,0 +1,430 @@
...
+          \ <Attribute Name=\"http://schemas.xmlsoap.org/ws/2005/05/identity/claims/name\"\
+          \ NameFormat=\"urn:oasis:names:tc:SAML:2.0:attrname-format:uri\" FriendlyName=\"\
+          Name\" xmlns=\"urn:oasis:names:tc:SAML:2.0:assertion\"/>\n        <Attribute\
+          \ Name=\"http://schemas.xmlsoap.org/ws/2005/05/identity/claims/surname\"\
+          \ NameFormat=\"urn:oasis:names:tc:SAML:2.0:attrname-format:uri\" FriendlyName=\"\
+          Surname\" xmlns=\"urn:oasis:names:tc:SAML:2.0:assertion\"/>\n        <Attribute\
+          \ Name=\"http://schemas.xmlsoap.org/ws/2005/05/identity/claims/nameidentifier\"\
+          \ NameFormat=\"urn:oasis:names:tc:SAML:2.0:attrname-format:uri\" FriendlyName=\"\
+          Name ID\" xmlns=\"urn:oasis:names:tc:SAML:2.0:assertion\"/>\n      </IDPSSODescriptor>\n\
+          \    </EntityDescriptor>\n    "
```
Comment (melvyndekort) [inline casc_configs/jenkins.yaml:216]: You've broken every line into a separate string, but can't you use the literal style like you did at the bottom of this file?

File: casc_configs/jenkins.yaml [hunk 2]
```diff
@@ -0,0 +1,430 @@
...
+          http://www.w3.org/2000/09/xmldsig#\">\n            <X509Data>\n        \
+          \      <X509Certificate>MIIC8jCCAdqgAwIBAgIJSsIGsbP+i6cpMA0GCSqGSIb3DQEBBQUAMCAxHjAcBgNVBAMTFXBvcnRiYXNlLmV1LmF1dGgwLmNvbTAeFw0xNjA5MDUwODAyMDFaFw0zMDA1MTUwODAyMDFaMCAxHjAcBgNVBAMTFXBvcnRiYXNlLmV1LmF1dGgwLmNvbTCCASIwDQYJKoZIhvcNAQEBBQADggEPADCCAQoCggEBAMS+LxZ4z36UfnGvIsFfCoZbRGjl7GZhgmADlcWBzj10qp4WYGio/ST3dNQUASUXGdnbREZ86ckJs/c1GlGfLzziHLxk9ncYiakjy8gUQOOnwiwbsWykiMmX2UpnTuuoZWV5br14fXLyjY233v4vtsIBzhc8SBr01ItBVNX9zTtc7ZaLwoAg9MB0aI3jo6BlkasUJOFJkxPNK5PRTedyKviv1PSZKbM7cXAwJAWx0TmmASJAQMDXusBFurtN5XG63lPt9xwThb7XwTctU0ayLWHhk1ew/VN1YcDiQXKt0fdI3TmyhJ24mG0EBOZo/lPMdx61O1xL3tkUERZXwIApIwECAwEAAaMvMC0wDAYDVR0TBAUwAwEB/zAdBgNVHQ4EFgQUUXeayGHj6ouuZ5zZeDwmkqVecpQwDQYJKoZIhvcNAQEFBQADggEBABNjN9HAM6WaIA77dGKPdYFCGrlm+npC6p0ES5aVjaJYe6Lon6vm5nnwnbQ9k+f12iiqjMKNzwoy6bkePXDd4LB/ODmny1j0mnPXwmSLna3PQWgop+19mJZaAaVzZLpXtKEsHe2DcnEAH+MsfWuTKg1pJjF0tU/0GBXO1LGQnmt16LK4qlyZlvFrNg2By35rQVn7KafCf/YDJJ1BIKGnKOlWPbhxSI3czuh3Ym379XkmxmC2ZUufnZVvlVYBtz0+gatgsZX0r4mofTPAKD3Jdny0v1zKS+wmX6QhpxBmIo0CSGEfd171V3107lkVJq8pXlqXs1KwgXP54e/g33cEycI=</X509Certificate>\n\
```
Comment (melvyndekort) [inline casc_configs/jenkins.yaml:188]: Isn't this secure data that should never have been published to this repo???

File: casc_configs/jenkins.yaml [hunk 3]
```diff
@@ -0,0 +1,430 @@
+jenkins:
```
Comment (melvyndekort) [inline casc_configs/jenkins.yaml:1]: - Ik zie de credentials niet terugkomen in deze YAML file, klopt dat?

File: casc_configs/jenkins.yaml (hunk 1 above)
Comment (r-kok) [reply casc_configs/jenkins.yaml]: this fragment is an export of the CASC plugin itself, I did not choose the format

File: casc_configs/jenkins.yaml (hunk 2 above)
Comment (r-kok) [reply casc_configs/jenkins.yaml]: obscure data. will cycle anyway

File: casc_configs/jenkins.yaml (hunk 3 above)
Comment (r-kok) [reply casc_configs/jenkins.yaml:1]: yes

**Pull request merged**